lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -666,8 +664,52 @@ def infer(self, images: List[Image.Image], threshold: float) -> List[List[Elemen
                 )
                 elements.append(element)
             batched_results.append(elements)
+            if self.cache:
+                key = Cache.get_hash_key(image.tobytes())
+                self.cache.set(key, result)
+
         return batched_results
 
+    def _get_cached_inference(self, images: List[Image.Image], threshold: float) -> list:
+        results = []
+        uncached_images = []
+        uncached_indices = []
+
+        # First, check the cache for each image
+        for index, image in enumerate(images):
+            key = Cache.get_hash_key(image.tobytes())
+            cached_image = self.cache.get(key)
+            if cached_image:
+                print("Cache hit. Returning the result from cache")
```

Comment:
What would be nicer here would be a cache hit-rate from self.cache.  We know what a cache hit is.  We don't need explanation that we're using the result from the cache.

Reply:
updated